staging/src/k8s.io/client-go/tools/cache/thread_safe_store.go

```diff
@@ -70,6 +71,15 @@ type threadSafeMap struct {
 	indices Indices
 }
 
+func (c *threadSafeMap) Error(err error) {
+	errKey := "error"
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	oldObject := c.items[errKey]
+	c.items[errKey] = err
+	c.updateIndices(oldObject, err, errKey)
+}
```

Comment:
hmm... this looks weird... can we not propagate the error in a stateless way without it touching the store or indexes?